src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,149 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+
+    private Entry<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Entry[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        if (key == null) {
+            putForNullKey(value);
+            return;
+        }
+
+        int hash = hash(key);
+        int index = indexFor(hash, table.length);
```

Comment:
you do not need hash in this method. You need only index.

```suggestion
        int index = indexFor(key);
```